src/index.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Represents the object returned by the Checkout creation API call.
+ * @see https://api.prod.easypay.pt/docs#tag/Checkout/paths/~1checkout/post
+ */
+interface CheckoutManifest {
+  id: string
+  session: string
+  config: null | {
+    language?: string
+    logoUrl?: string
+    openBtn?: string
+    closeBtn?: string
+    primaryButtons?: string
+    secondaryButtons?: string
+    logoImg?: string
+    stepBadgePrimary?: string
+    stepBadgeSecondary?: string
+    stepTextPrimary?: string
+    stepTextSecondary?: string
+    cartBtn?: string
+    cart?: string
+    cartItem?: string
+    hideDetails?: boolean
+  }
+}
+
+/**
+ * The possible options to configure the Checkout SDK.
+ */
+interface CheckoutOptions {
+  /** The id of the HTML element in which to insert the Checkout form. */
+  id?: string
+  /** The callback to call on Checkout events. */
+  onMessage?: (message: string) => void
+  /** Whether the SDK should use testing APIs. */
+  testing?: boolean
+}
+
+/** The default values for all the options. */
+const defaultOptions: CheckoutOptions = {
+  id: 'easypay-checkout',
+  onMessage: () => { /* do nothing */ },
+  testing: false,
+}
+
+const LOGTAG = '[easypay Checkout SDK]'
+
+/** Validates the necessary parameters for Checkout initialization and gives helpful messages for integrators. */
+function validateParameters(manifest: CheckoutManifest, options: CheckoutOptions) {
+  if (!manifest || typeof manifest.id !== 'string' || manifest.id === '' || typeof manifest.session !== 'string' || manifest.session === '') {
+    console.error(`${LOGTAG} Please provide a valid Checkout Manifest when calling startCheckout.`)
+    return false
+  }
+  if (typeof options.id !== 'string' || options.id === '') {
+    console.error(`${LOGTAG} The HTML element id must be a non-empty string.`)
+    return false
+  }
+  if (typeof options.onMessage !== 'function') {
+    console.error(`${LOGTAG} The onMessage callback must be a function.`)
+    return false
+  }
+  return true
+}
+
+/** Encodes a Manifest for iframe consumption. Supports both Node and browser environments. */
+function encodeManifest(manifest: CheckoutManifest) {
+  if (typeof Buffer !== 'undefined') {
+    return Buffer.from(JSON.stringify(manifest), 'binary').toString('base64')
+  } else {
+    return btoa(JSON.stringify(manifest))
+  }
+}
+
+/** Used to configure and populate the Checkout form. */
+export function startCheckout(manifest: CheckoutManifest, options: CheckoutOptions = {}) {
+  const mergedOptions = {...defaultOptions, ...options}
+  if (!validateParameters(manifest, mergedOptions)) {
+    return
+  }
+  let url = 'https://checkout-serverless.quality-utility.aws.easypay.pt'
```

Comment:
I would create constants for these magic values.

Reply:
Ok. They're also bound to change, correct? We'll need one for production to substitute the "quality" one, which will only be available via direct URL parameter, is that right?